lib/puppet/util/log.rb

```diff
@@ -336,7 +336,7 @@ def level=(level)
   # If they pass a source in to us, we make sure it is a string, and
   # we retrieve any tags we can.
   def source=(source)
-    if source.respond_to?(:path)
+    if defined?(Puppet::Type) && source.is_a?(Puppet::Type)
```

Comment:
That is from the `is_resource?` method of Puppet::Util::Logging at https://github.com/puppetlabs/puppet/blob/3.8.1/lib/puppet/util/logging.rb#L197-L199 , but this log util may not be used before `require 'puppet/type'` though the Puppet::Util::Logging one may, but I included it to be safe.